### PR TITLE
docs: update catalyst references

### PR DESCRIPTION
**Pull request by mikemurray**
Impact: **minor**  
Type: **docs**

## Solution
- Update docs in various places
- Update dev dependencies to resolve lint and test issues
- Update tests
- Add resolution for `lodash` security vuln
- Updated the snyk ignore for the unfixable dev dependency vuln for the `mem` package

## Testing
1. Read through the docs and ensure that references to the `component-library` are replaced with `catalyst`.
1. All tests should be passing on CI, and there should be a preview site published for this PR.

### CODE REVIEW

**Comment by mikemurray** [comment]
@machikoyasuda This is ready for review

File: README.md [hunk 1]
```diff
@@ -1,6 +1,6 @@
 # Reaction Component Library & Design System
 
-[![npm (scoped)](https://img.shields.io/npm/v/@reactioncommerce/components.svg)](https://www.npmjs.com/package/@reactioncommerce/catalyst)
+[![npm (scoped)](https://img.shields.io/npm/v/@reactioncommerce/catalyst.svg)](https://www.npmjs.com/package/@reactioncommerce/catalyst)
  [![CircleCI](https://circleci.com/gh/reactioncommerce/catalyst.svg?style=svg)](https://circleci.com/gh/reactioncommerce/catalyst)
 
 ![Reaction Design System](https://blog.reactioncommerce.com/content/images/2018/09/style-guide-artwork.jpg)
```
Comment (machikoyasuda) [inline README.md]: Line 11:

- [Catalyst Design System](https://catalyst.reactioncommerce.com/): See the root [package.json](https://github.com/reactioncommerce/catalyst/blob/master/package.json).

Comment (machikoyasuda) [inline README.md]: Add this to Line 7 for now: (we can remove it when we actually _release_ this)

> ⚠️ Warning: This repository is in active development and is not ready for testing or usage.

Line 8:

This is a single project with a package of operator-focused React UI components and code for the Catalyst Design System website:

**Comment by machikoyasuda** [comment]
Approved from my side. @mikemurray merge when you're ready.

**Pull request merged**